classes/class-p4-campaigns.php

```diff
@@ -44,6 +49,10 @@ public function __construct() {
 		 * Class hooks.
 		 */
 		private function hooks() {
+
+			add_action( 'init', [ $this, 'register_campaigns_cpt' ] );
+			add_action( 'add_meta_boxes',  [ $this, 'campaign_page_templates_meta_box' ] );
+			add_action( 'save_post', [ $this, 'save_campaign_page_templates_meta_box_data' ] );
```

Comment:
Although the hook contains many checks, I would suggest using a hook that will run only for the specific custom post type.
https://developer.wordpress.org/reference/hooks/save_post_post-post_type/